go-controller/pkg/libovsdbops/db_object_types.go

```diff
@@ -38,6 +39,16 @@ const (
 
 // ObjectIDsTypes should only be created here
 
+var AddressSetAdminNetworkPolicy = newObjectIDsType(addressSet, AdminNetworkPolicyOwnerType, []ExternalIDKey{
+	// anp name
+	ObjectNameKey,
+	// egress or ingress
+	PolicyDirectionKey,
+	// gress rule's priority
+	PriorityKey,
```

Comment:
> I think it may be easier to use gress rule index here instead of priority. We have changed that some time ago for network policies too, thinking that db indexes should refer to the k8s object fields rather than implementation details (which is how we evaluate priority). It should also be more clear that every gress rule has its own address set in that case (same applies to acls).
> 
> In the ANP case, this will allow us to not delete/recreate objects when ANP priority changes, because address set index wouldn't change at all, and acl will be update of the existing row instead of recreating with the new IDs. Constructing db predicates hopefully will also become somewhat easier. Does it make sense?

maybe yes, I am ok with doing the "indices way" instead i.e use `GressIdxKey` instead of `PriorityKey` but let's make sure this is the final way forward because earlier I was deleting and recreating the address-sets always, then we realized we want to just delete the ones we care about and update the rest which led to the predicates we have today, but note that the predicates will now just become the indices calculation instead of priority calculation. Currently I am not deleting and recreating all address-sets (NewAdressSetOps is using CreateOrUpdateAddressSetsOps which is always just updating the externalIDs and setting the newIPs not creating new address-sets), this logic will stay the same except that we will stop updating the externalIDs {if we don't want to track PriorityKey} and only update the IPs in the address-set.

> Oh, the name is not mandatory. What a pity!

yeah rule names are unfortunately not mandatory and not unique so not really considering `name` or rules and `name` of acls for any of the main calc logic..

Reply:
this is done, I have changed to using indices instead of priority ranges and that means like Nadia said, we can remove the updates we need to do to address-sets if priority changed..
same for acls, i have moved to using indices instead of priority...